Frisp/Initialiser.php

```
@@ -257,11 +257,17 @@ protected function initBucket($config) {
 			echo "\n\n";
 		}
 
-		// And upload to S3
-		$s3Client->putBucketPolicy([
-			'Bucket' => $this->env->bucket,
-			'Policy' => json_encode($policy, JSON_PRETTY_PRINT)
-		]);
+
+		// yet another eventually consistency check, wait 30 secs and then bail if the
+		// Aws\S3\Exception\S3Exception - "Invalid principal in policy" persists
+		$bucket = $this->env->bucket;
+		$this->tryIgnoringException('Aws\S3\Exception\S3Exception', function() use($s3Client, $bucket, $policy) {
```

I can see if there is a special exception code or in worst case do string matching? in worst case scenario it will only ignore the exception for 30 seconds.

Apparently the s3 api isn't as verbose when it comes to different exception types.